src/components/Tasks.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { v4 } from 'uuid'
+// import { addtask, removeTask, updateTask } from '../actions'
+import { ADDTASK, REMOVETASK, UPDATETASK } from '../redux/tasks'
+import './tasks.css'
+const Tasks = () => {  
+  const tasks=useSelector((state)=>state.tasks)
+  console.log(tasks)
+  const dispatch=useDispatch()
+  const uid=v4()
+  const handleEditButton=(t)=>{
+    seteditinput(t)
+    dispatch(UPDATETASK({...t,isEdited:true}))
+  }
+  const [input,setinput]=useState({
+    id:uid,
+    task:'',
+    completed:false,
+    isEdited:false
+  })
+  const [editinput,seteditinput]=useState({
+    id:uid,
+    task:'',
+    completed:false,
+    isEdited:false
+  })
+
+  const dispatchFunction=()=>{
+    dispatch(ADDTASK(input))
+    setinput((prev)=>{
+        return{
+            ...prev,
+            id:v4(),
+        }
+    })
+  }
+  return (
+    <div className='task-container'>
+        <h1>Task Todo List</h1>
+    <div className='task-input'>
+        <input type="text" 
+        value={input.task}
+        onChange={(e)=>{
+            setinput((prev)=>{
+                return{
+                    ...prev,
+                    task:e.target.value
+                }
```

Comment:
Avoid compound states like this. Here you should have a separate state for the task.

Reply:
It’s better to have more states than a single compound state.